### PR TITLE
config: setter/getter for Notifier and Logger into its own struct.

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
This is an attempt cleanup code and keep the top level config
functions simpler and easy to understand where as move the
notifier related code and logger setter/getter methods as part
of their own struct.

Locks are now held properly not globally by configMutex, but
instead as private variables.

## Motivation and Context
Final fix for #3700

## How Has This Been Tested?
Cleanup

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/notifiers.go [hunk 1]
```diff
@@ -66,12 +68,128 @@ type notifier struct {
...
 
+// SetAMQPByID -
+func (n *notifier) SetAMQPByID(accountID string, amqpn amqpNotify) {
+	n.Lock()
+	defer n.Unlock()
+	n.AMQP[accountID] = amqpn
+}
+
+func (n *notifier) GetAMQP() map[string]amqpNotify {
+	return n.AMQP
```
Comment (vadmeste) [inline cmd/notifiers.go]: Can we hold a lock here and return a copy of AMQP ? the same of other types of notification.

Comment (vadmeste) [reply cmd/notifiers.go]: LGTM otherwise

Comment (harshavardhana) [reply cmd/notifiers.go]: Why would you hold a lock ? We are not modifying the members.

Comment (vadmeste) [reply cmd/notifiers.go]: > Why would you hold a lock ? We are not modifying the members.

Currently, the caller of GetAMQP will iterate over n.AMQP elements, we can get a problem with a concurrent write

Comment (harshavardhana) [reply cmd/notifiers.go]: You are right memory is same.

**Review by vadmeste — APPROVED** [review]

**Pull request merged**